qa/rolling-upgrade/src/test/java/org/elasticsearch/upgrades/RecoveryIT.java

```diff
@@ -796,15 +795,15 @@ public void testAutoExpandIndicesDuringRollingUpgrade() throws Exception {
                 Settings.builder().put(IndexMetaData.INDEX_ROUTING_EXCLUDE_GROUP_PREFIX + "._id", nodes.get(randomInt(2))));
         }
 
-        ensureGreen(indexName);
-
-        final int numberOfReplicas = Integer.parseInt(
-            getIndexSettingsAsMap(indexName).get(IndexMetaData.SETTING_NUMBER_OF_REPLICAS).toString());
-        if (minimumNodeVersion.onOrAfter(Version.V_7_6_0)) {
-            assertEquals(nodes.size() - 2, numberOfReplicas);
-        } else {
-            assertEquals(nodes.size() - 1, numberOfReplicas);
-        }
+        assertBusy(() -> {
+            final int numberOfReplicas = Integer.parseInt(
+                getIndexSettingsAsMap(indexName).get(IndexMetaData.SETTING_NUMBER_OF_REPLICAS).toString());
+            if (minimumNodeVersion.onOrAfter(Version.V_7_6_0)) {
```

Comment:
Do we need the `assertBusy()` if `minimumNodeVersion.onOrAfter(Version.V_7_6_0)` or can we `ensureGreen()` in that case?

Reply:
Thanks for the question. We actually don't need assertBusy at all (https://github.com/elastic/elasticsearch/pull/30423 was already introduced in 6.4, I misremembered this).

The `ensureGreen` is mostly for cosmetics. I've added it for the case where we are on 7.6.0+